src/Security/Authorization/Policy/src/AuthorizationMiddlewareOptions.cs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Authorization
+{
+    /// <summary>
+    /// Options for an <see cref="AuthorizationMiddleware"/>.
+    /// </summary>
+    public class AuthorizationMiddlewareOptions
+    {
+        /// <summary>
+        /// When an <see cref="AuthorizationHandler{TRequirement}"/> is invoked as part of authorization in the <see cref="AuthorizationMiddleware"/>,
+        /// the <see cref="AuthorizationHandlerContext.Resource"/> instance defaults to being the <see cref="Endpoint"/> that will be processed.
+        /// <para>
+        /// When <see langword="true"/>, the <see cref="AuthorizationMiddleware"/> instead configures the <see cref="AuthorizationHandlerContext.Resource"/> to be
+        /// an instance of <see cref="AuthorizationMiddlewareContext"/>. This allows handlers to access the current request context without the need to use
+        /// <see cref="IHttpContextAccessor"/>.
+        /// </para>
+        /// </summary>
+        public bool AllowRequestContextInHandlerContext { get; set; }
```

Comment:
Do we want to consider just adding this to `AuthorizationOptions` https://github.com/aspnet/AspNetCore/blob/master/src/Security/Authorization/Core/src/AuthorizationOptions.cs

The middleware is already effectively relying on the `FallbackPolicy` that is set there, although that is more of a sideeffect of how Combine works, so its not quite an apples to apples comparison

Reply:
@HaoK we could. The only caveat is that callsite is this is configuring the callsite. In MVC's case without endpoint routing (and I think SignalR), this option really doesn't have any effect. So putting it something as "global" as `AuthorizationOptions` feels a bit icky. I'm going to run this by the API review, it'll be good to get a couple more opinions.